### ISSUE TITLE
`@stylistic/declaration-block-trailing-semicolon` declines a fix that would land past an inline comment, not inside it

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#287]
Found by the review of the branch closing [#232](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/232), and it arrives with that branch: before it, this rule asked a declaration nothing at all.

### What minimal example or steps are needed to reproduce the bug?

```less
a {
	--x: pink // c
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "customSyntax": "postcss-less",
  "rules": {
    "@stylistic/declaration-block-trailing-semicolon": "always"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.less" --config myconfig.json --fix
```

### What did you expect to happen?

The semicolon written, and nothing else touched:

```less
a {
	--x: pink // c
;}
```

Which is what the rule does today, and what it stops doing.

### What actually happened?

The problem is reported and no fix is offered, though the semicolon has a line of its own to stand on. `postcss-scss` reads such a value the same way, so the shape accounts for six of the twenty-seven rows a sweep of this rule moves — three under each syntax, one for each spelling of the inline comment the sweep is written with. Under `postcss-scss` those three are the whole of what moves: everywhere else that syntax either keeps the comment in the raw the rule guarded already, as it does behind the parameters of a bodiless at-rule, or reads it as a node of its own, which leaves the declaration out of the rule's reach.

### What is the cause?

`endsWithInlineComment` trims the trailing whitespace of the text it is handed, and says so: whatever a fixer writes goes where that whitespace is, the line break closing the comment among it, so none of it counts as closing anything. That is the question every caller but one asks. `declaration-block-semicolon-space-before` and `declaration-block-semicolon-newline-before` write over the trailing whitespace of a value, and `function-parentheses-newline-inside` says as much in a comment of its own, working around the trim by writing the parenthesis back on the end before it asks.

`declaration-block-trailing-semicolon` asks the opposite question. Its fix sets `parent.raws.semicolon`, which puts the semicolon behind the whole printed text of the node, trailing whitespace and all. So a line break the node's own text ends with does close the comment as far as this write goes, and the trim erases the one character that proves it.

A custom property is where the two part, since both parsers read its value to the end of the line and take the break into the value. Everywhere else a declaration's value stops in front of the break, the write really would land inside the comment, and the guard is right.

`writesIntoInlineComment` has no way to be told which of the two writes it is standing in front of. Giving it one — a second argument, or a second name — is the fix; picking the text by hand at the call site is not, since that is how [#211](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/211) came about.
